src/key/factory.js

```diff
@@ -37,20 +37,15 @@ import { unarmor } from '../encoding/armor';
 
 /**
  * Generates a new OpenPGP key. Supports RSA and ECC keys.
- * Primary and subkey will be of same type.
- * @param {module:enums.publicKey} [options.keyType=module:enums.publicKey.rsaEncryptSign]
- *                             To indicate what type of key to make.
- *                             RSA is 1. See {@link https://tools.ietf.org/html/rfc4880#section-9.1}
- * @param {Integer} options.rsaBits    number of bits for the key creation.
- * @param {String|Array<String>}  options.userIds
- *                             Assumes already in form of "User Name <username@email.com>"
- *                             If array is used, the first userId is set as primary user Id
- * @param {String}  options.passphrase The passphrase used to encrypt the resulting private key
- * @param {Number} [options.keyExpirationTime=0]
- *                             The number of seconds after the key creation time that the key expires
- * @param  {String} options.curve            (optional) elliptic curve for ECC keys
- * @param  {Date} options.date         Override the creation date of the key and the key signatures
- * @param  {Array<Object>} options.subkeys   (optional) options for each subkey, default to main key options. e.g. [{sign: true, passphrase: '123'}]
+ * By default, primary and subkeys will be of same type.
+ * @param {ecc|rsa} options.type              The primary key algorithm type: ECC or RSA
+ * @param {String}  options.curve             Elliptic curve for ECC keys
+ * @param {Integer} options.rsaBits           Number of bits for RSA keys
+ * @param {Array<String>}  options.userIds    Array of user IDs e.g. [{ name:'Phil Zimmermann', email:'phil@openpgp.org' }]
```

Comment:
The type contradicts the example (the type says string but the example is an object). In fact the type can be `String|Object|Array<String>|Array<Object>`, though perhaps that's a bit unwieldy :upside_down_face:
(Same below)

Reply:
I think we require an array here, only generateKey takes a single string/object.